AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/IgnoreWhenHandler.java

```diff
@@ -53,6 +59,21 @@ public void process(Element element, EFragmentHolder holder) throws Exception {
 		JMethod delegatingMethod = codeModelHelper.overrideAnnotatedMethod(executableElement, holder);
 		JBlock previousMethodBody = codeModelHelper.removeBody(delegatingMethod);
 
-		delegatingMethod.body()._if(invoke(holder.getGeneratedClass().staticRef("this"), "getActivity").ne(_null()))._then().add(previousMethodBody);
+		IgnoreWhen ignoreWhen = element.getAnnotation(IgnoreWhen.class);
+		switch (ignoreWhen.value()) {
+		case VIEW_DESTROYED:
+			String viewDestroyedFieldName = "viewDestroyed" + generationSuffix();
```

Comment:
We usually do this with the holder. Add a new field the holder class, which is initialised and related code is created on demand (lazily, when we call `holder.getViewDestroyedField()`).